orttraining/orttraining/models/runner/training_runner.cc

```diff
@@ -1068,11 +1096,42 @@ Status TrainingRunner::Evaluate(InferenceSession& session, IDataLoader& data_loa
                                 fetch_names,
                                 fetches);
 
-    ORT_RETURN_IF_ERROR(session.Run(run_options,
-                                    feed_names,
-                                    feeds,
-                                    fetch_names,
-                                    &fetches));
+    auto status = Status::OK();
+    if (params_.pipeline_parallel_size == 1) {
```

Comment:
This `if-else` choose different ways to launch `session.Run(...)`. If there is no pipeline, all `session.Run(...)` are delegated to the first thread. If pipeline is enabled, all threads are occupied by pipeline stages and we should not reuse any of them.